web/openapi/openapi.yaml

```diff
@@ -16,6 +16,73 @@ components:
       in: header
       name: X-VAST-Token
 paths:
+  /query:
+    post:
```

Comment:
Have you considered making it a GET?

Pro: The request doesn't change the state of VAST [ref](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)
Pro: Easier to refresh and share.  Alternatively, the svelte frontend could have the query in GET parameters while still querying the backend API through a POST.
Cons: Would limit the size of the queries (but queries are small in VAST?). For example on the Splunk setup I know, the search is passed as a GET parameter, at least on the frontend (not sure if different for the backend).

Reply:
Are there downsides when supporting both GET and POST? Then we can let the user choose their preferred method.

Reply:
GET would require random access iterator semantics if I understand correctly, while POST just requires input iterator semantics. This ultimately depends on whether we want to keep data sets in memory, or apply some deterministic ordering to them (currently, there is no order between events from different partitions, just within a single partition).

Reply:
Ah I didn't consider or know about the random order, I think it would be an argument for the POST instead of GET.
In terms of web expectations I think you'd use GET if the output is always the same for the same data (so I think you would want to enforce a default ordering over different partitions).
For the limit of the URL it's 2040 characters [ref](https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers), but again Splunk doesn't seem to run into the issue so maybe VAST wouldn't either.

In practical use cases I would find it convenient to be able to just save a search by copy-pasting the URL, which wouldn't be possible if neither the backend or frontend support a GET for /query, or would require short URL-generating code (ala Google flight, booking.com, etc.) which would probably something in the backend and a database to keep the short URL mappings.

Reply:
I think I've now understood the workflow proposed: The user would make a `/query` request to start the search and then `/query/<id>` to see the results?
In that case I think both use cases with POST and GET would make sense. Some users would want to always see the same data as a result, some others (like what I described) would want to save the search but not freeze the result data.

Reply:
Yeah, the workflow is:
1. Come with an expression to `/query` to create a query, get back an ID
2. Plow through `/query/{id}` in a forward-iterator fashion, i.e., a one-time sweep through the result set, where you can provide the chunking (n).

For (1), POST or GET don't matter that much. But for (2), we don't want GET right now because there is no idempotence: a refresh will show the *next* batch of results for the provided `n`.

To put this into perspective: we plan to have a fully deterministic "dataset" abstraction in the future. This will then support GET so that you can share the URL and get all the idempotent properties you'd expect when refreshing a page.

Reply:
Thanks, I think it's what was confusing me. I think I get (😉) it now and it makes sense, thanks all for your patience and the great work.